controllers/subscriber/subscriberControllers.md

```diff
@@ -0,0 +1,175 @@
+
+Once this has been set up, components can subscribe to particular registries two different ways:
+1. Using a matching event name with `EventSubscriberController`. The component will need to be a child of the registry component for this to work.
+2. By pointing to the registry component's id with `IdSubscriberController`. The component will need to be in the same DOM scope as the registry component for this to work.
+
+Like the `SubscriberRegistryController`, these `*subscriberController`s take optional callbacks to throw at different points in the subscription process.
+
+```js
+import { EventSubscriberController, ForPropertySubscriberController } from '@brightspace-ui/core/controllers/subscriber/subscriberControllers.js';
+
+class AnimalFactsUI extends LitElement {
+	static get properties() {
+		return {
+			catFact: { type: String },
+			dogFact: { type: String }
+		};
+	}
+
+	constructor() {
+		super();
+		this._catSubscriberController = new EventSubscriberController(this,
+			{ onSubscribe: this._catsRejoice.bind(this) }
+			{ eventName: 'd2l-interesting-fact-subscribe-cat', controllerId: 'cat' }
+		);
+
+		this._dogSubscriberController = new EventSubscriberController(this, {},
+			{ eventName: 'd2l-interesting-fact-subscribe-dog', controllerId: 'dog' }
+		);
+	}
+
+	render() {
+		return html`
+			<p>Cat Fact: ${this.catFact}</p>
+			<p>Dog Fact: ${this.dogFact}</p>
+		`;
+	}
+
+	...
+}
+
+class GeneralFactsUI extends LitElement {
+	static get properties() {
+		return {
+			for: { type: String },
+			_fact: { type: String }
+		};
+	}
+
+	constructor() {
+		super();
+		this._generalSubscriberController = new IdSubscriberController(this,
+			{ onError: this._onError.bind(this) },
+			{ idPropertyName: 'for', controllerId: 'general' }
+		);
+	}
+
+	render() {
+		return html`
+			<p>Did you know? ${this._fact}</p>
+		`;
+	}
+
+	setFact(fact) {
+		this._fact = fact;
+	}
+
+	_onError() {
+		throw new Error('We need facts!');
+	}
+
+	...
+}
+```
+
+An example of what this could look like altogether:
+```html
+<interesting-fact-subscription id="facts">
+	<animal-facts-ui></animal-facts-ui>
+</interesting-fact-subscription>
+<general-facts-ui for="facts"></general-facts-ui>
+```
+
+NOTE: Until we are on Lit 2, the controller lifecycle events will need to be manually called:
+```js
+	connectedCallback() {
+		super.connectedCallback();
+		if (this._subscriptionController) this._subscriptionController.hostConnected();
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._subscriptionController) this._subscriptionController.hostDisconnected();
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (this._subscriptionController) this._subscriptionController.hostUpdated(changedProperties);
+	}
+```
```

Comment:
I found the sample code a little hard to follow in terms of tying what the example was trying to do to the core concepts. 🐶  🐱  It's very cute though and I did learn fun facts about tongue prints!

Maybe a metaphor that also uses a subscription pattern could work? Like a "newspaper" <-> "reader" or a "cable company" <-> "viewer" type metaphor? Do people still even subscribe to newspapers?!

Reply:
Yeah I adapted this from the Provider/Requester README, but using a more intuitive subscription example makes sense.  I'll try it with cable channels and viewers

Reply:
Ok, updated - let me know if that works better. It's still a bit of a weird example when looking at the resulting html, but I wanted to show how event subscribers must go inside the registry component, and id subscribers can go outside.

Reply:
I like this a lot! 👍